modules/dnn/src/layers/resize2_layer.cpp

```diff
@@ -0,0 +1,1010 @@
+            {
+                int idy = iy + k;
+                unsigned valid = (unsigned)idy < (unsigned)inH;
+                y_id[oy][k+1] = valid ? idy : -1;
+                float w = y_w[oy][k+1];
+                float wv = valid ? w : 0.f;
+                y_w[oy][k+1] = wv;
+                swy += wv;
+                hasOutOfBounds |= !valid;
+            }
+        }
+        else
+        {
+            for (int k = -1; k <= 2; ++k)
+            {
+                int idy = iy + k;
+                unsigned valid = (unsigned)idy < (unsigned)inH;
+                y_id[oy][k+1] = valid ? idy : -1;
+                float w = y_w[oy][k+1];
+                float wv = valid ? w : 0.f;
+                y_w[oy][k+1] = wv;
+                swy += wv;
+            }
+        }
+        if (swy != 0.f)
+            for (int k = 0; k < 4; ++k)
+                y_w[oy][k] /= swy;
+        if (tf_crop_and_resize_mode && hasOutOfBounds) {
+            outOfBoundsY[oy] = true;
+        }
+    }
+
+    const int nstripes = 16;
+    parallel_for_(Range(0, nstripes), [&](const Range& range) {
+        int row0 = range.start * (outH * numPlanes) / nstripes;
+        int row1 = range.end   * (outH * numPlanes) / nstripes - 1;
+        int plane0 = row0 / outH, plane1 = row1 / outH;
+        row0 %= outH;
+        row1 %= outH;
+
+        for (int p = plane0; p <= plane1; ++p)
+        {
+            int oy0 = (p == plane0) ? row0 : 0;
+            int oy1 = (p == plane1) ? row1 : outH - 1;
+            const float* inpBase = inpPlanes.ptr<float>(p * inH);
+            for (int oy = oy0; oy <= oy1; ++oy)
+            {
+                float* outRow = outPlanes.ptr<float>(p * outH) + oy * outW;
+
+                if (tf_crop_and_resize_mode && outOfBoundsY[oy]) {
+                    for (int ox = 0; ox < outW; ++ox)
+                        outRow[ox] = extrapolation_value;
+                    continue;
+                }
+
+                std::vector<float> hbuf(inW, 0.f);
+                for (int ix = 0; ix < inW; ++ix)
+                {
+                    float v = 0.f;
+                    bool hasValid = false;
+                    for (int ky = 0; ky < 4; ++ky)
+                    {
+                        int yy = y_id[oy][ky];
+                        if (yy < 0) continue;
+                        const float* row = inpBase + yy * inW;
+                        v += y_w[oy][ky] * row[ix];
+                        hasValid = true;
+                    }
+                    hbuf[ix] = hasValid ? v : 0.f;
+                }
+
+                for (int ox = 0; ox < outW; ++ox)
+                {
+                    if (tf_crop_and_resize_mode && outOfBoundsX[ox]) {
+                        outRow[ox] = extrapolation_value;
+                        continue;
+                    }
+                    float val = 0.f;
+                    bool hasValid = false;
+                    for (int kx = 0; kx < 4; ++kx)
```

Comment:
you always access 4 subsequent locations: x-1, x, x+1, x+2. Why do you keep all 4 values inside the table? Also, check for inliners first, similar to what you did in gridSample implementation.